---
title: "R Notebook"
output: html_notebook
---
```{r}
library(tidyverse)
library(ggplot2)
library(readxl)
library(reshape2)
library(phyloseq)
library(microbiome)
library(ggpubr)
library(RColorBrewer)
library(viridis)
```
#Task1 and Task2 are creating the right bracken data that match metadata with genotype and vivo data. But they are still 10 samples which is not in the bracken data.
#Task3 is making the phyloseq object
##Task1
###Combine the metadata (having both genotype and vivo gas data) with bracken data and create a new file with only abundance data

##Task2
###Combine the metadata (having both genotype and vivo gas data) with bracken data and create a new file with both abundance and count data_all_combined_bracken

##Task3

```{r}
#copy the 
#set the working directory
setwd("/ebio/abt3_projects/hubif_lactose_intolerance/hubif_lactose_desk/fin_analysis/noexpats")
#Import OTU table without taxonomy,Taxonomy table and sample metadata
OTU = read.csv("bracken_reads_noexpats.csv", header = TRUE)
tax = read.csv("taxonomy.csv", header = TRUE)
#check the column name of the taxonomy table
colnames(tax)
#seperate the taxnomy be different levels
tax.sep = separate(tax, taxonomy, into = c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), sep=";")
# removing the name and taxonomy_id columns since these are now row names
#fix(tax.clean)
str(tax.sep)
write.csv(tax.sep,file = "tax_sep.csv")
#Edit the tax_sep table by paste the OTU name column from OTU Table in the first column, save as tax_sep_edit
```
#12 samples doesn't have bracken data, so we need to remove them from metadata
```{r}
setwd("/ebio/abt3_projects/hubif_lactose_intolerance/hubif_lactose_desk/fin_analysis/noexpats")
#Import OTU table without taxonomy,Taxonomy table and sample metadata
OTU = read.csv("bracken_reads_noexpats.csv", header = TRUE)
meta = read.csv("meta_noexpats.csv", header = TRUE)
f <- colnames(OTU)[-1]
meta_sel <- meta %>% subset(ParticipantID %in% f)
dim(meta)
dim(meta_sel)
setwd("/ebio/abt3_projects/hubif_lactose_intolerance/hubif_lactose_desk/fin_analysis/noexpats")
write.csv(meta_sel, "meta_noexpats_sel.csv")

setwd("/ebio/abt3_projects/hubif_lactose_intolerance/hubif_lactose_desk/fin_analysis/noexpats")
tax = read.csv("tax_sep_edit.csv", header = TRUE)
meta = read.csv("meta_noexpats_sel.csv", header = TRUE)
#check if OTU table and meta have the same sample size
meta$ParticipantID == colnames(OTU)[-1]


row.names(OTU) = OTU$name
#removing the column otu since it is now as a row name
OTU.clean = OTU %>% select(-name)
# doing the same for the other matrixes
row.names(tax) <- tax$name
tax.clean <- tax %>% select(-name)
row.names(meta) <- meta$sample
meta.clean <- meta %>% select(-sample)
#transmoring into matrixes otu table and tax tables
OTU.clean <- as.matrix(OTU.clean)
tax.clean <- as.matrix(tax.clean)
#Transforming into phyloseq objects
OTU = otu_table(OTU.clean, taxa_are_rows = TRUE)
TAX = tax_table(tax.clean)
samples = sample_data(meta.clean)
#create the phyloseq object
physeq = phyloseq(OTU, TAX, samples)
physeq
summarize_phyloseq(physeq)
```
Saving phyloseq object to RDS
```{r}
saveRDS(physeq, "/ebio/abt3_projects/hubif_lactose_intolerance/hubif_lactose_desk/fin_analysis/noexpats/phyloseq/physeq.rds")
```
###Comparison of the Chao1 and Shannon index of populations among 3 different continents
```{r}
#Rarefying 
seed_num= 7460 #keep this number to ensure reproducibility
rarefy_depth_1 = 1000000
physeq_100k = rarefy_even_depth(physeq,  sample.size=rarefy_depth_1, rngseed=seed_num, replace=FALSE)
# prune OTUs that are not present in at least one sample (AFP = Africa_pruned)
AFP <- prune_taxa(taxa_sums(physeq_100k) > 0, physeq_100k)
AFP
#set the alpha diveristy you want to plot. Chao1-estimate diveristy from abundance data(count of different species/OTUs in each sample, importance of rare OTUs). Shannon - evenness of the sample
alpha_meas = c("Chao1", "Shannon")
#alpha_meas = c("Chao1", "Shannon", "Simpson", "InvSimpson")
#pairwise comparisons, the default method is "wilcox.test". 
my_comparisons <- list(c("Gabon", "Vietnam"), c("Gabon", "Germany"), c("Vietnam", "Germany"))
#plot the a-diversity in different continents
p1 <- plot_richness(AFP, "Country", measures=alpha_meas) + geom_boxplot(aes(x=Country, y=value, color=Country), alpha=0.01)  + 
  scale_x_discrete(limits = c("Gabon", "Vietnam", "Germany")) +
  stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  theme_bw() + 
  theme(axis.text.x = element_text(angle = 0)) +
#remove the background of the boxplot
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank()) + theme(aspect.ratio = 1.5) +
xlab("") +
    scale_y_continuous(expand = expansion(mult = c(0.05, 0.08)))
#+ ggtitle("Microbiome Diverisity in different Countries")
#  theme(legend.position = "None") +
p1
```
```{r}
p = plot_richness(AFP, x="Country", color="Country", measures=alpha_meas)
p + theme(plot.title = element_text(hjust = 0.5), legend.position = "none") +
    geom_violin() +
     geom_boxplot(width=0.2) +
    scale_x_discrete(limits = c("Germany", "Gabon", "Vietnam")) + 
  stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 0)) +
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank()) + theme(aspect.ratio = 1.5) +
xlab("") +
  theme(legend.position = "None")

annotate_figure(p, bottom = text_grob("Rarefied to lowest common depth \n ITM Means ± SD = Observed: 861 ± 233; Shannon: 4.6 ± 0.68 ", color = "blue", hjust = 1, x = 1, face = "italic", size = 10))

```


###Family-level gut microbiome composition of populations in 3 different continents. 
```{r}
physeqobject = AFP

wunifrac_dist1 = phyloseq::distance(physeqobject, method="bray", weighted=F)
ordination1 = ordinate(physeqobject, method="PCoA", distance=wunifrac_dist1)
  plot1 <- plot_ordination(physeqobject, ordination1, color="Country") + theme(aspect.ratio=1) + ggtitle("Bray Curtis") + theme(plot.title = element_text(hjust = 0.5)) + theme_bw()
plot1 + geom_point(size=0.1, alpha=0.1) 

wunifrac_dist2 = phyloseq::distance(physeqobject, method="jaccard", weighted=T)
ordination2 = ordinate(physeqobject, method="PCoA", distance=wunifrac_dist2)
plot2 <- plot_ordination(physeqobject, ordination2, color="Country") + theme(aspect.ratio=1) + ggtitle("Jaccard") + theme(plot.title = element_text(hjust = 0.5)) + theme_bw() + geom_point(size=0.1, alpha=0.1)
plot2


figure <- ggarrange(plot1, plot2, common.legend=TRUE, legend="bottom")
annotate_figure(figure, 
                top = text_grob("Dissimilarity Matrices", size = 18, face = "bold"),
                bottom = text_grob("Rarefaction: 1M reads (same used for PAM clustering)", color = "blue",
                                   hjust = 1, x = 1, face = "italic", size = 10))

ggsave('/ebio/abt3_projects/hubif_lactose_intolerance/hubif_lactose_desk/fin_analysis/genotype_vivo_620/phyloseq/Dissimilarity.pdf')
```
```{r}
physeqobject = AFP

wunifrac_dist1 = phyloseq::distance(physeqobject, method="bray", weighted=T)
ordination1 = ordinate(physeqobject, method="PCoA", distance=wunifrac_dist1)
  plot1 <- plot_ordination(physeqobject, ordination1, color="Country") + theme(aspect.ratio=1) + ggtitle("Bray Curtis") + theme(plot.title = element_text(hjust = 0.5)) + theme_bw() + geom_point(size=0.1, alpha=0.1) 
plot1

wunifrac_dist2 = phyloseq::distance(physeqobject, method="jaccard", weighted=F)
ordination2 = ordinate(physeqobject, method="PCoA", distance=wunifrac_dist2)
plot2 <- plot_ordination(physeqobject, ordination2, color="Country") + theme(aspect.ratio=1) + ggtitle("Jaccard") + theme(plot.title = element_text(hjust = 0.5)) + theme_bw() 
plot2


figure <- ggarrange(plot1, plot2, common.legend=TRUE, legend="bottom")
annotate_figure(figure, 
                top = text_grob("Dissimilarity Matrices", size = 18, face = "bold"),
                bottom = text_grob("Rarefaction: 1M reads", color = "blue",
                                   hjust = 1, x = 1, face = "italic", size = 10))

ggsave('/ebio/abt3_projects/hubif_lactose_intolerance/hubif_lactose_desk/fin_analysis/genotype_vivo_620/phyloseq/Dissimilarity.pdf')
```
###Microbiota alpha- and beta diverisity from 3 populations in different continents. (A) alpha diveristy (B, C) show PCoA plots illustrating beta-diversity based on (B) Jccard diveristy, and (C) Unweighted Unifrac distance. The first component explained nearly ?% based on the Jaccard diverity of the individual variation and ?% based on the Unweighted Unifrac distance. 


```{r}

#Look into taxa abundance change by family level
family <- AFP %>%
  tax_glom(taxrank = "Family") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.02) %>%                         # Filter out low abundance taxa
  arrange(Family)
nrow(family)
My_theme = theme(
  axis.title.x = element_text(size = 10),
  axis.title.y = element_text(size = 10),
  axis.text.x = element_text(size = 8, hjust= 0.5, vjust= 0.5),
  axis.text.y = element_text(size = 8),
  plot.title = element_text(size=10))

# Create a ggplot with 18 colors 
nb.cols <- 65
mycolors <- colorRampPalette(brewer.pal(12, "Paired"))(nb.cols)

theme_update(plot.title = element_text(hjust = 0.5))
t <- ggplot(family, aes(x = Country, y = Abundance, fill = Family)) + 
  geom_bar(stat = "identity", position = "fill") +
  ylab("Relative Abundance") +
  xlab("Condition") +
  ggtitle("Microbiome composition") +
  theme(axis.text.x = element_text(angle = 0)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) +
  theme(plot.title = element_text(hjust = 0.5))+
  theme(legend.text=element_text(size=7), legend.title=element_text(size=7)) +
# Use scale_fill_manual
    scale_fill_manual(values = mycolors) +
  theme(legend.position="left") + theme(aspect.ratio = 1) +
  My_theme

t

ggsave('/ebio/abt3_projects/hubif_lactose_intolerance/hubif_lactose_desk/fin_analysis/genotype_vivo_620/phyloseq/Composition at family level.pdf')
```


```{r}
nb.cols <- 43
mycolors <- colorRampPalette(brewer.pal(15, "Dark2"))(nb.cols)

theme_update(plot.title = element_text(hjust = 0.5))
ggplot(family, aes(x = Country, y = Abundance, fill = Family)) + 
  geom_bar(stat = "identity", position = "fill") +
  ylab("Relative Abundance") +
  xlab("Condition") +
  ggtitle("Microbiome composition at family level in different continents") +
    theme(plot.title = element_text(hjust = 0.5))+
  theme(axis.text.x = element_text(angle = 0)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) +
  theme(legend.text=element_text(size=7), legend.title=element_text(size=7)) +
# Use scale_fill_manual
    scale_fill_manual(values = mycolors) +
  theme(legend.position="right") + theme(aspect.ratio = 2) +
  My_theme

```

```{r}
#Look into taxa abundance change by family level
genus <- AFP %>%
  tax_glom(taxrank = "Genus") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.05) %>%                         # Filter out low abundance taxa
    arrange(Genus)
nrow(genus)
My_theme = theme(
  axis.title.x = element_text(size = 10),
  axis.title.y = element_text(size = 10),
  axis.text.x = element_text(size = 8, hjust= 0.5, vjust= 0.5),
  axis.text.y = element_text(size = 8),
  plot.title = element_text(size=10))

genus
# Create a ggplot with 18 colors 
nb.cols <- 82
mycolors <- colorRampPalette(brewer.pal(12, "Paired"))(nb.cols)

ggplot(genus, aes(x = Clade1_imputed, y = Abundance, fill = Genus)) + 
  geom_bar(stat = "identity", position = "fill") +
  ylab("Relative Abundance") +
  xlab("Condition") +
  ggtitle("Microbiome composition at genus level in different continents") +
  theme(axis.text.x = element_text(angle = 0)) +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.background = element_blank(), axis.line = element_line(colour = "black")) +
  theme(plot.title = element_text(hjust = 0.5)) +
  theme(legend.text=element_text(size=5), legend.title=element_text(size=7)) +
# Use scale_fill_manual
    scale_fill_manual(values = mycolors) +
  theme(legend.position="left") + theme(aspect.ratio = 2) +
  My_theme
```





